loopchain/tools/config_gen/helper.py

```diff
@@ -0,0 +1,158 @@
+    mktree.make_tree()
+
+    return "0x" + mktree.get_merkle_root().hex()
+
+
+def make_channel_config(crep_root_hash: str, radiostations: list = None, **kwargs) -> Config:
+    channel_config = {
+        "block_versions": {
+            "0.1a": 0,
+            "0.3": 1
+        },
+        "hash_versions": {
+            "genesis": 1,
+            "0x2": 1,
+            "0x3": 1
+        },
+        "load_cert": False,
+        "consensus_cert_use": False,
+        "tx_cert_use": False,
+        "key_load_type": 0,
+        "radiostations": radiostations,
+        "crep_root_hash": crep_root_hash
+    }
+
+    for key, value in kwargs.items():
+        channel_config[key] = value
+
+    return channel_config
+
+
+# ----- Peers
+def make_peer_config(channels_config: Config, port_peer: int, key: Key, **kwargs) -> Config:
+    peer_config = dict()
+    peer_config["CHANNEL_OPTION"] = channels_config
+    peer_config["PRIVATE_PATH"] = str(key.path)
+    peer_config["PRIVATE_PASSWORD"] = key.password
+    peer_config["RUN_ICON_IN_LAUNCHER"] = True
+    peer_config["PORT_PEER"] = port_peer
+
+    for key, value in kwargs.items():
+        peer_config[key] = value
+
+    return peer_config
+
+
+# ----- Channel Manage Data
+def _make_peers_in_channel_manage_data(keys: Keys) -> List[Config]:
+    peers = []
+
+    for order, key in enumerate(keys):
+        grpc_port = make_port_by_order(order, start_port=PORT_PEER)
+        peer = {
+            "id": key.address,
+            "peer_target": f"[local_ip]:{grpc_port}",
+            "order": order + 1
+        }
+        peers.append(peer)
+
+    return peers
+
+
+def make_channel_manage_data(channel_names: list, keys: Keys) -> Config:
+    channel_manage_data = {}
+    for channel_name in channel_names:
+        channel_manage_data[channel_name] = {
+            "score_package": "score/icx",  # TODO: Not needed?
+            "peers": _make_peers_in_channel_manage_data(keys=keys)
+        }
+
+    return channel_manage_data
+
+
+# ----- Other helper funcs
+def make_port_by_order(order: int, start_port: int) -> int:
+    return start_port + order * PORT_DIFF_BETWEEN_PEERS
+
+
+def make_radiostations_for_local(n: int) -> list:
+    """Make radiostations for local test."""
+    return [f"[local_ip]:{PORT_PEER_FOR_REST + (i * PORT_DIFF_BETWEEN_PEERS)}"
+            for i in range(n)]
+
+
+def check_param_exist(attr_name: str):
+    """Decorator on class methods that checks that its instance attr exists."""
+    def _check(func):
+        def __check(self, *args, **kwargs):
+            if getattr(self, attr_name) is None:
+                raise RuntimeError(f"build_{attr_name} first")
+            else:
+                return func(self, *args, **kwargs)
+        return __check
+    return _check
+
+
+def dict_write(path: Union[Path, str], dict_obj: dict):
+    """Write dict value to path."""
+    if os.path.exists(path):
+        raise FileExistsError(f"{path} already exists.")
```

Comment:
What if I want to regenerate config files with different number of total reps? Should I remove the directory before `loop init`?

Reply:
Currently, yes. You have to remove configure root folder first.

---

Suppose that you generate 8 peers. Then you need to decrease the number of peer as 4. So you generate again 4 peers.
> In this case, if the config root folder exists, you will notice that [0-7]th peer configs are in root folder. [0-3] are new one, and [4-7] are old one.

I wanted to avoid this confusing situation. 

---

Do you any good idea??